### PR TITLE
Flg optimize chunk generation

### PR DESCRIPTION
**Pull request by filedesless**
Replace nested vec by a larger flat vec to save on multiple heap allocations

### CODE REVIEW

**Comment by afrigon** [comment]
build is failing @filedesless

File: src/chunk/chunk.rs [hunk 1]
```diff
@@ -5,22 +5,91 @@ pub const CHUNK_WIDTH: usize = 16;
 pub const CHUNK_DEPTH: usize = 16;
 pub const CHUNK_HEIGHT: usize = 256;
 
-type Blocks = Vec<Vec<Vec<Block>>>;
+const CHUNK_SIZE: usize = CHUNK_WIDTH * CHUNK_DEPTH * CHUNK_HEIGHT;
+
+type Blocks = Vec<Block>;
+
+/// computes flat array index from a 3d coordinate using pre-computed lookup tables
```
Comment (filedesless) [inline src/chunk/chunk.rs]: fix comment

File: src/chunk/chunk.rs [hunk 2]
```diff
@@ -5,22 +5,91 @@ pub const CHUNK_WIDTH: usize = 16;
...
+fn at(x: usize, y: usize, z: usize) -> usize {
```
Comment (afrigon) [inline src/chunk/chunk.rs:14]: on pourrait probablement gagner du speed en mettant la directive #[inline] ici et sur get_block/set_block

**Comment by afrigon** [comment]
> build is failing @filedesless

we could gate the test crate behind a feature to still be able to compile on the stable toolchain, like mentioned in this post.

https://www.reddit.com/r/learnrust/comments/aqp97b/featuregating_featuretest/

**Review by afrigon — APPROVED** [review]

**Pull request merged**